src/util/xml_to_json.h

```diff
@@ -45,22 +45,22 @@ class Xml2Json {
 public:
     class Hints {
     public:
-        void setArrayName(const pugi::xml_node& node, const std::string& name) { asArray[node] = name; }
-        void setFieldType(const std::string& node, const std::string& type) { asType[node] = type; }
+        void setArrayName(const pugi::xml_node& node, std::string_view name) { asArray[node] = name; }
+        void setFieldType(std::string_view node, std::string_view type) { asType[node] = type; }
 
     private:
-        std::map<pugi::xml_node, std::string> asArray;
-        std::map<std::string, std::string> asType;
+        std::map<pugi::xml_node, std::string_view> asArray;
+        std::map<std::string_view, std::string_view> asType;
```

Comment:
Keep these as strings, to create copies. `string_view` stores raw pointers which can become invalid.